// ng-plyr/esm2020/lib/services/ng-plyr.service.mjs
import { Injectable } from '@angular/core';
import { ErrorCodes } from '../models/error.model';
import { Playlist } from '../models/playlist.model';
import * as i0 from "@angular/core";
export class PlayerService {
    constructor() { }
    _checkForErrors(elements) {
        let check = new Set(elements);
        if (!this._plyr)
            throw new Error(ErrorCodes['player/not-initialized']);
        if (check?.has('video') && !this._plyr.video)
            throw new Error(ErrorCodes['video/element-not-initialized']);
    }
    // Accessing `this` from player component
    addComponentRef(plyrComponent) {
        this._plyr = plyrComponent;
    }
    removeComponentRef() {
        this._plyr = undefined;
    }
    // Public functions
    // Actions
    play() {
        this._checkForErrors(['video']);
        this._plyr.video?.nativeElement.play();
    }
    pause() {
        this._checkForErrors(['video']);
        this._plyr.video?.nativeElement.pause();
    }
    next() {
        this._checkForErrors();
        this._plyr.playNextMedia();
    }
    prev() {
        this._checkForErrors();
        this._plyr.playPrevMedia();
    }
    enableMediaLooping(loop = true) {
        this._checkForErrors();
        this._plyr.isLoopingEnabled = loop;
    }
    enablePlaylistLooping(loop = true) {
        this._checkForErrors();
        this._plyr.loopPlaylist = loop;
    }
    changeVolume(level) {
        this._checkForErrors(['video']);
        this._plyr.changeVolume(level);
    }
    seekTo(atSecond) {
        this._checkForErrors(['video']);
        this._plyr.seekTo(atSecond);
    }
    setPlaybackSpeed(rate) {
        this._checkForErrors();
        this._plyr.setPlaybackSpeed(rate);
    }
    // Getter functions
    getCurrentlyPlaying() {
        this._checkForErrors();
        return this._plyr.media;
    }
    getNextMedia() {
        this._checkForErrors();
        return this._plyr.nextMedia;
    }
    getNumOfMediaInPlaylist() {
        this._checkForErrors();
        return this._plyr.playlist.itemCount;
    }
    // Modify playlist
    addToPlaylist(mediaItems, atStart) {
        this._checkForErrors();
        let items = Playlist.mediaArrToPaylistItems(mediaItems);
        this._plyr.playlist.appendPlaylist(items, atStart);
    }
    playNext(media) {
        this._checkForErrors();
        let items = Playlist.mediaArrToPaylistItems([...media]);
        this._plyr.playlist.addNext(items);
    }
}
PlayerService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PlayerService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
PlayerService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PlayerService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PlayerService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return []; } });
