// ng-plyr/esm2020/lib/ng-plyr.module.mjs
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { NgPlyrComponent } from './ng-plyr.component';
import { StopClickPropagationDirective } from './directive/stop-click-propagation.directive';
import * as i0 from "@angular/core";
export class NgPlyrModule {
}
NgPlyrModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NgPlyrModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgPlyrModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: NgPlyrModule, declarations: [NgPlyrComponent,
        StopClickPropagationDirective], imports: [BrowserModule], exports: [NgPlyrComponent] });
NgPlyrModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NgPlyrModule, imports: [BrowserModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NgPlyrModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgPlyrComponent,
                        StopClickPropagationDirective
                    ],
                    imports: [
                        BrowserModule
                    ],
                    exports: [
                        NgPlyrComponent
                    ]
                }]
        }] });
